File: Smack/Transitions/DismissModalTransitionAnimator.m
//
//  DismissModalTransitionAnimator.m
//  cabcenter
//
//  Created by Jose Molina Colmenero on 12/06/2017.
//  Copyright © 2017 Frank Lindner. All rights reserved.
//

#import "DismissModalTransitionAnimator.h"

@implementation DismissModalTransitionAnimator

- (NSTimeInterval) transitionDuration:(id<UIViewControllerContextTransitioning>)transitionContext {
    return DEFAULT_ANIMATION_DURATION;
}

- (void) animateTransition:(id<UIViewControllerContextTransitioning>)transitionContext {
    UIViewController *fromViewController = [transitionContext viewControllerForKey:UITransitionContextFromViewControllerKey];
    
    [transitionContext.containerView addSubview:fromViewController.view];
    UIView *firstView = fromViewController.view.subviews[0];
    CGRect finalFrame = CGRectMake(firstView.frame.origin.x, fromViewController.view.frame.size.height, firstView.frame.size.width, firstView.frame.size.height);
    
    [UIView animateWithDuration:DEFAULT_ANIMATION_DURATION
                          delay:0
         usingSpringWithDamping:0.8
          initialSpringVelocity:0.1
                        options:UIViewAnimationOptionCurveEaseOut
                     animations:^{
                         fromViewController.view.backgroundColor = [UIColor clearColor];
                         firstView.frame = finalFrame;
                     } completion:^(BOOL finished) {
                         [transitionContext completeTransition:YES];
                     }];
}

@end

File: Smack/Transitions/DismissModalTransitionAnimator.swift
//
//  DismissModalTransitionAnimator.swift
//  Smack
//
//  Created by Jose Molina Colmenero on 28/07/2017.
//  Copyright © 2017 Jonny B. All rights reserved.
//

import UIKit

class DismissModalTransitionAnimator : NSObject, UIViewControllerAnimatedTransitioning {
    
    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
        return 0.5
    }
    
    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
        let fromViewController = transitionContext.viewController(forKey: .from)!
        transitionContext.containerView.addSubview(fromViewController.view)
        let firstView = fromViewController.view.subviews[0]
        let secondView = fromViewController.view.subviews[1]
        let finalFrame = CGRect(x: secondView.frame.origin.x,
                                y: fromViewController.view.frame.size.height,
                                width: secondView.frame.size.width,
                                height: secondView.frame.size.height)
        
        UIView.animate(withDuration: 0.5,
                       delay: 0.0,
                       options: .curveEaseOut,
                       animations: {
                        firstView.backgroundColor = UIColor.clear
                        secondView.frame = finalFrame
        }) { (finished) in
            transitionContext.completeTransition(true)
        }
    }

    
}
